File: files/Launch into Computing July 2025 B/Unit 12 - Part A/chatbot.js
// Chatbot in Javascript 
const readline = require('readline');

// Create the interface for reading input and displaying output
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

console.log("Chatbot: Hello there im your helper, Type 'bye' to exit.");

// Event listener for the line input
rl.on('line', (input) => {
  const msg = input.toLowerCase();  // Convert user input to lowercase

  // Check for different types of responses
  if (msg === 'bye') {
    console.log("Chatbot: Goodbye!");
    rl.close();  // Close the interface when the user types "bye"
  } else if (msg.includes('hello')) {
    console.log("Chatbot: Hi there!");
  } else if (msg.includes('how are you')) {
    console.log("Chatbot: For more advanced answers, contact Amnon my creator!");
  } else {
    console.log("Chatbot: I don't understand that.");
  }
});

File: files/Launch_into_Computing_July_2025/Unit_12_A/chatbot.py
# Chatbot in Python

def chatbot():
    # Introduction message
    print("Chatbot: Hello there im your helper, Type 'bye' to exit.")
    
    # Start an infinite loop to keep the conversation going
    while True:
        # Get user input and convert it to lowercase for easier comparison
        msg = input("You: ").lower()

        # Check for different types of responses
        if msg == 'bye':
            print("Chatbot: Goodbye!")
            break  # Exit the loop if the user types "bye"
        elif 'hello' in msg:
            print("Chatbot: Hi there!")
        elif 'how are you' in msg:
            print("Chatbot: For more advanced answers, contact Amnon my creator!")
        else:
            print("Chatbot: I don't understand that.")

if __name__ == "__main__":
    chatbot()  # Run the chatbot
